app/services/image_filename_service.rb

```diff
@@ -1,15 +1,22 @@
 # frozen_string_literal: true
 
 class ImageFilenameService
+  MAX_LENGTH = 65
+
   attr_reader :document
 
   def initialize(document)
     @document = document
   end
 
   def call(suggested_name, mime_type)
-    normalised_name = normalise(suggested_name, mime_type)
-    ensure_unique(normalised_name)
+    filename = ActiveStorage::Filename.new(suggested_name)
+
+    base = filename.base.parameterize.slice 0...MAX_LENGTH
+    base = ensure_unique(base)
+
+    extension = Rack::Mime::MIME_TYPES.invert[mime_type]
+    "#{base}#{extension}"
```

Comment:
Is there a missing `.` here, between base and extension?

Reply:
Weirdly the dot's included - I thought it would be more confusing to manually take it out and then manually put it in again :-|.